# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [47]:
NAME: 'Gabriele Pelucchi, 2010541'

ERROR: Error in eval(expr, envir, enclos): oggetto "NAME" non trovato


In [48]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [49]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1       V2        V3        V4        V5        V6        V7       
1 0.126339 0.9526136 0.2026399 0.7248014 0.9222768 0.4339289 0.4856934
2 0.100325 0.7846530 0.6689427 0.9907341 0.7792939 0.8151871 0.1018514
  V8        V9        V10       V11       V12       V13       V14      
1 0.6986412 0.4745053 0.3084830 0.4875614 0.8298769 0.7620575 0.2119169
2 0.4735931 0.2620462 0.5907095 0.7676283 0.2578982 0.3555699 0.1937899
  V15        V16       V17       V18       V19       V20      
1 0.05453112 0.6573601 0.5061291 0.9608215 0.1916136 0.9748142
2 0.41246961 0.2293659 0.6180676 0.5588683 0.8196640 0.3167716

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [50]:
df2 <- df1 %>%
     gather(Key,Value)
  

head(df2)

Key Value    
1 V1  0.1263390
2 V1  0.1003250
3 V2  0.9526136
4 V2  0.7846530
5 V3  0.2026399
6 V3  0.6689427

## Assignment 2

We will generate some data:

In [51]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [52]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [53]:
df5 <- full_join(df3,df4,by='id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook